### DISCUSSION TITLE
Disable blinking cursor on mouseleave

### DISCUSSION THREAD

**Discussion by tilman-schieber** [jcubic/jquery.terminal#900] [Q&A] [answered]
I am having a lot of fun with the terminal, it works great together with pyodide to embed a python console in a course I develop. 
However, as there is other stuff going on on the page, I would like to disable the blinking cursor when the mouse is not over the terminal window. 
If this is not feasible, I would also be fine with disabling the blinking cursor altogether, 
I tested `--animation: terminal-none;` but this makes the cursor completely invisible. 

I tried to make sense of /css/jquery.terminal.css but the animations to achieve the blinking cursor are way over my head.

**Comment by jcubic** [comment, marked as answer]
`--animation: terminal-none` disables animation, not the cursor, I'm not sure how the cursor can be invisible. Do you have a reproduction that shows that cursor disappear?

I think that would you need is to disable the terminal on the mouse out and enable it on hover.

Here is the code you can use:

```javascript
$(document).on('mouseover', '.terminal', (e) => {
    $(e.currentTarget).terminal().enable();
}).on('mouseout', '.terminal', (e) => {
    $(e.currentTarget).terminal().disable();
});
```

And here is [a demo](https://codepen.io/jcubic/pen/YzdZaaY?editors=0110).

**Reply by tilman-schieber** [reply]
Your enable/disable solution is exactly what I needed. 
Thank you for your helpful reply.

Having a non-blinking cursor would be fine as well, but I cannot get it to work. 
Please don't feel obligated to debug my mess but here's the code example that shows no cursor for me at all: 
https://gist.github.com/tilman-schieber/380b7b03c869939bfded4c7065ccacf7

**Reply by jcubic** [reply]
The problem is that the `none` animation was added in version [2.33.0](https://github.com/jcubic/jquery.terminal/blob/master/CHANGELOG.md#2330) and you have 2.23.0.

**Reply by tilman-schieber** [reply]
🤦‍♂️
thank you. 
I upgraded to 2.35.2 and it works now.
